src/urllib3/response.py

```diff
@@ -455,6 +457,9 @@ def data(self):
         if self._fp:
             return self.read(cache_content=True)
 
+    def json(self) -> Mapping[str, str]:
```

Comment:
I think this should be `-> Any`, because nothing restricts the response to only JSON objects.

Also, I think a docstring would be nice here, for the API reference. In particular, I would mention the exception this can raise (`json.JSONDecodeError`); since this is user-controlled input, a lot of users of this function would want to handle it.